### PR TITLE
Expose upload feature

### PR DESCRIPTION
**Pull request by Kryptos-FR**
Split the upload feature into two: start and poll similar to the "get data" feature.

See also https://github.com/fragcolor-xyz/chainblocks-dotnet/pull/6

### CODE REVIEW

File: library/src/lib.rs [hunk 1]
```diff
@@ -115,50 +127,105 @@ pub extern "C" fn clmrPollFree(state: *mut PollState) {
...
 }
 
-pub fn proto_upload(
+pub fn start_proto_upload(
   node_url: &str,
   signer_key: &str,
   auth_key: &str,
   proto_type: &str,
   data: Table,
-) -> Result<(), &'static str> {
```
Comment (sinkingsugar) [inline library/src/lib.rs:124]: No this is not the way, we want to keep pure rust calls outside silly Box (extra mallocs)

Comment (sinkingsugar) [reply library/src/lib.rs:124]: I suggest extract this and add an helper for boxed version like I did for download

Comment (Kryptos-FR) [reply library/src/lib.rs:124]: Not sure what you mean. That's exactly what I did. I mimicked the way download worked.

Comment (sinkingsugar) [reply library/src/lib.rs:124]: hah you are right... I implemented it the way I said before and did not work.. so had to Box :) my bad....

**Comment by sinkingsugar** [comment]
good to go for me

**Pull request merged**